app/controllers/AdditionalMetadataNavigationController.scala

```diff
@@ -32,20 +33,34 @@ class AdditionalMetadataNavigationController @Inject() (
       .toList
       .filter(_ != "csrfToken")
       .map(UUID.fromString)
-    if (metadataType == "closure") {
-      val fileFilters = FileFilters(None, Option(fileIds), None)
-      consignmentService
-        .getConsignmentFileMetadata(consignmentId, request.token.bearerAccessToken, Option(fileFilters))
-        .map(consignment => {
-          val areAllClosed = consignmentService.areAllFilesClosed(consignment)
-          if (areAllClosed) {
-            Redirect(routes.AddClosureMetadataController.addClosureMetadata(consignmentId, fileIds))
-          } else {
-            Redirect(routes.AdditionalMetadataClosureStatusController.getClosureStatusPage(consignmentId, fileIds))
-          }
-        })
+
+    if (fileIds.nonEmpty) {
```

Comment:
This nested if else statement is getting a bit long and deep. Wondering if its abstracting out a bit into a separate function to handle the different views for "closure" and "descriptive". Something like `selectView`?

Reply:
Yeah, that makes sense. I moved to a new method.